web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php

```diff
@@ -238,6 +240,86 @@ public function nodeEntityPresave(EntityInterface $entity): void {
     }
   }
 
+  /**
+   * Find overdue proposals.
+   *
+   * @return array|int
+   *   A list of overdue proposals
+   */
+  public function findOverdueProposals() {
+    $now = new DrupalDateTime();
+    try {
+      return $this->entityTypeManager
+        ->getStorage('node')
+        ->getQuery()
+        ->condition('status', 1, '=')
+        ->condition('field_content_state', 'active', '=')
+        ->condition('field_vote_end', $now->format(DateTimeItemInterface::DATE_STORAGE_FORMAT), '<')
+        ->execute();
+    }
+    catch (\Exception $exception) {
+      $this->logger->error('Error finding overdue proposals: @message', [
+        '@message' => $exception->getMessage(),
+      ]);
+    }
+
+  }
+
+  /**
+   * Finish a proposal by changing it's state and som other fields.
+   *
+   * @param int $nid
+   *   The id of the proposal to finish.
+   */
+  public function finishProposal(int $nid): void {
+    try {
+      $entity = $this->entityTypeManager
+        ->getStorage('node')
+        ->load($nid);
+
+      if (!$entity instanceof Node) {
+        return;
+      }
+
+      // Set proposal state.
+      $entity->set('field_content_state', 'finished');
+
+      // @todo Use method to determine support count when it is merged.
+      $supportCount = $this->connection->select('hoeringsportal_citizen_proposal_support')
+        ->condition('node_id', $entity->nid->value ?? 0)
+        ->countQuery()
+        ->execute()
+        ->fetchField();
+
+      // @todo Use constant to compare against when it is merged.
+      // Anonymize proposal if it didn't reach enough votes.
+      if ($supportCount < 10) {
```

Comment:
Consider adding a `hasEnoughVotes()`, say, function to this class to keep the code clean.